config/routes.rb

```diff
@@ -0,0 +1,16 @@
+Rails.application.routes.draw do
+  get 'users', to: 'users#index', as: 'users'
+  get 'users/:id',to: 'users#show', as: 'user'
+  get 'login', to: 'sessions#new', as: 'login_form'
+  post '/login', to: 'sessions#create', as: 'login'
+  delete '/logout', to: 'sessions#destroy', as: 'logout'
+
+  resources :votes
```

Comment:
Do you need all the routes here for votes?  I think you should only have a route to create a vote, maybe nested under `publications`  You wouldn't need a route for `index`, or `new`